### PR TITLE
[7.x] Escape attributes automatically in some situations

### PR DESCRIPTION
**Pull request by taylorotwell**

### CODE REVIEW

File: src/Illuminate/View/Compilers/ComponentTagCompiler.php [hunk 1]
```diff
@@ -165,15 +176,15 @@ protected function componentString(string $component, array $attributes)
         if (! class_exists($class)) {
             $parameters = [
                 'view' => "'$class'",
-                'data' => '['.$this->attributesToString($data->all()).']',
+                'data' => '['.$this->attributesToString($data->all(), $escapeBound = false).']',
```
Comment (perifer) [inline src/Illuminate/View/Compilers/ComponentTagCompiler.php]: This is currently not covered by any test (Changing to `$escapeBound = true` doesn't cause any tests to fail).

Comment (perifer) [reply src/Illuminate/View/Compilers/ComponentTagCompiler.php]: Here is an attempt to cover this in `testClasslessComponents()`: https://github.com/perifer/framework/commit/46176c631faff707a5545ffb48199c8fb349690e

Comment (taylorotwell) [reply src/Illuminate/View/Compilers/ComponentTagCompiler.php]: Thanks

**Comment by GrahamCampbell** [comment]
* https://blog.laravel.com/security-laravel-712-released
* https://github.com/laravel/framework/releases/tag/v7.1.2

**Comment by bakerkretzmar** [comment]
@perifer @taylorotwell @GrahamCampbell is there any way to disable this sanitizing if I _do_ want to pass in a class with a `__toString()` method, and have it available as a PHP object (not a string) inside my component?

**Pull request merged**